Karl/integrations/nodejs/karl_server.cc

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2016 Maxim Zhurovich <zhurovich@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#define EXTRA_KARL_LOGGING
+#include "current_build.h"
+
+#include "../../karl.h"
+#include "../../../Bricks/dflags/dflags.h"
+
+DEFINE_uint16(karl_keepalives_port, 30001, "Local test port for the `Karl` keepalive listener.");
+DEFINE_uint16(karl_fleet_view_port, 30999, "Local test port for the `Karl` fleet view listener.");
+DEFINE_string(karl_stream_persistence_file, ".current/stream", "Local file to store Karl's keepalives.");
+DEFINE_string(karl_storage_persistence_file, ".current/storage", "Local file to store Karl's status.");
+
+CURRENT_STRUCT(custom_status) {
+  CURRENT_FIELD(custom_field, std::string);
+  void Render(std::ostream & os, std::chrono::microseconds) const {
+    os << "<TR><TD COLSPAN='2'>"
+       << "Custom JS Claire field value: " << custom_field << "</TD></TR>";
```

Comment:
Btw, custom_field should probably be htmlencoded before outputting in the HTML context.